filters/speech-tag/filter.py

```diff
@@ -0,0 +1,133 @@
+import operator
+import spacy
+
+from interfaces.SentenceOperation import SentenceOperation
+from tasks.TaskTypes import TaskType
+from collections import defaultdict
+from typing import Union
+
+"""
+A filter on if the tokens contain specific speech tag a certain number of times.
+"""
+
+
+class SpeechTagFilter(SentenceOperation):
+    tasks = [TaskType.TEXT_CLASSIFICATION, TaskType.TEXT_TO_TEXT_GENERATION]
+    languages = ["en"]
+
+    def __init__(
+        self,
+        speech_tags: Union[list, str] = ["NOUN", "VERB"],
+        thresholds: Union[list, int] = [2, 2],
+        operations: Union[list, str] = [">", ">"],
+        percentages: bool = False,
+    ):
+        super().__init__()
+        self.max_input_length = self.get_input_length(
+            speech_tags, thresholds, operations
+        )
+        self.final_operators = self.parse_operator(operations)
+        self.final_speech_tags = self.convert_scalar_to_list(speech_tags)
+        self.final_thresholds = self.convert_scalar_to_list(thresholds)
+        self.nlp = spacy.load("en_core_web_sm")
```

Comment:
For optimization, we are loading spacy only once in 'initialize.py'. Please change this line with:
self.nlp = spacy_nlp if spacy_nlp else spacy.load("en_core_web_sm")

You can refer to other filters like 'keywords', 'length'.

Reply:
done